lib/jobs/message_latency_report.ex

```diff
@@ -123,15 +135,39 @@ defmodule Jobs.MessageLatencyReport do
         station: station
       ]
     end)
-    |> Enum.sort_by(fn row -> row[:seconds] end)
-    |> Enum.with_index()
+  end
+
+  defp filter_by_date(row, date) do
```

Comment:
I admit, this helper function is a little hacky, but it seems like Elixir doesn't have very rich support for parsing datetimes and this gets the job done.